gcloud-java-bigquery/src/main/java/com/google/gcloud/bigquery/QueryStage.java

```diff
@@ -0,0 +1,431 @@
+  private final long recordsRead;
+  private final long recordsWritten;
+  private final List<QueryStep> steps;
+  private final double waitRatioAvg;
+  private final double waitRatioMax;
+  private final double writeRatioAvg;
+  private final double writeRatioMax;
+
+  static final class Builder {
+
+    private double computeRatioAvg;
+    private double computeRatioMax;
+    private long id;
+    private String name;
+    private double readRatioAvg;
+    private double readRatioMax;
+    private long recordsRead;
+    private long recordsWritten;
+    private List<QueryStep> steps;
+    private double waitRatioAvg;
+    private double waitRatioMax;
+    private double writeRatioAvg;
+    private double writeRatioMax;
+
+    private Builder() {}
+
+    Builder computeRatioAvg(double computeRatioAvg) {
+      this.computeRatioAvg = computeRatioAvg;
+      return this;
+    }
+
+    Builder computeRatioMax(double computeRatioMax) {
+      this.computeRatioMax = computeRatioMax;
+      return this;
+    }
+
+    Builder id(long id) {
+      this.id = id;
+      return this;
+    }
+
+    Builder name(String name) {
+      this.name = name;
+      return this;
+    }
+
+    Builder readRatioAvg(double readRatioAvg) {
+      this.readRatioAvg = readRatioAvg;
+      return this;
+    }
+
+    Builder readRatioMax(double readRatioMax) {
+      this.readRatioMax = readRatioMax;
+      return this;
+    }
+
+    Builder recordsRead(long recordsRead) {
+      this.recordsRead = recordsRead;
+      return this;
+    }
+
+    Builder recordsWritten(long recordsWritten) {
+      this.recordsWritten = recordsWritten;
+      return this;
+    }
+
+    Builder steps(List<QueryStep> steps) {
+      this.steps = steps;
+      return this;
+    }
+
+    Builder waitRatioAvg(double waitRatioAvg) {
+      this.waitRatioAvg = waitRatioAvg;
+      return this;
+    }
+
+    Builder waitRatioMax(double waitRatioMax) {
+      this.waitRatioMax = waitRatioMax;
+      return this;
+    }
+
+    Builder writeRatioAvg(double writeRatioAvg) {
+      this.writeRatioAvg = writeRatioAvg;
+      return this;
+    }
+
+    Builder writeRatioMax(double writeRatioMax) {
+      this.writeRatioMax = writeRatioMax;
+      return this;
+    }
+
+    QueryStage build() {
+      return new QueryStage(this);
+    }
+  }
+
+  QueryStage(Builder builder) {
+    computeRatioAvg = builder.computeRatioAvg;
+    computeRatioMax = builder.computeRatioMax;
+    id = builder.id;
+    name = builder.name;
+    readRatioAvg = builder.readRatioAvg;
+    readRatioMax = builder.readRatioMax;
+    recordsRead = builder.recordsRead;
+    recordsWritten = builder.recordsWritten;
+    steps = builder.steps;
+    waitRatioAvg = builder.waitRatioAvg;
+    waitRatioMax = builder.waitRatioMax;
+    writeRatioAvg = builder.writeRatioAvg;
+    writeRatioMax = builder.writeRatioMax;
+  }
+
+  /**
+   * Returns the relative amount of time the average shard spent on CPU-bound tasks.
```

Comment:
Ah, the descriptions in https://cloud.google.com/bigquery/query-plan-explanation#available_metadata seemed better to me. Feel free to change or to keep
based on your preference.